libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/util/ServletRequestUtil.java

```diff
@@ -45,5 +46,20 @@ public static String getRemoteAddress(final HttpServletRequest request) {
         }
         return addr;
     }
+
+    /**
+     * Return the SIA user agent name, which is set by sia agent as request header.
+     * @param request http servlet request
+     * @return SIA agent along with version, like 'SIA-FARGATE 1.32.0'
+     **/
+    public static String getSiaAgent(final HttpServletRequest request) {
```

Comment:
is there any reason why we want to maintain the full value of the agent including the version number? Why not just extract the value after SIA- and then our provider will be EKS, AWS, Fargate, etc so it's easier to index and just run queries. Otherwise you'll have SIA-EKS 1.52.0, SIA-EKS 1.53.0, etc.

Reply:
With the provider agent version it will give more flexibility in general to filter based on certain agent version. Since the version is at the end queries can still use fuzzy search in indexes.
Also will be scalable if another type of Service Identity Agent come into picture in addition to SIA

Reply:
If that is the case, then the field should be called userAgent and not siaProvider since the user agent is not provider. Not sure exactly what use cases you're referring to regarding the version number - that information is already in the access log file so no need to maintain it here. In fact, for a lot of lambda identities, we're gong to get something like 'Jersey/2.35 (Apache HttpClient 4.5.13)' so that's not useful at all. 

We should keep our data clean and only save what we absolutely need. So we should get the agent, parse and only take agents with the format "SIA-<provider> <some other data>", extract the provider and save that value only. For all other cases, the provider will be empty and that should be excluded from any processing.